entrypoint.sh

```diff
@@ -19,6 +19,10 @@ else
     useradd -o --uid "$USER_ID" --gid "$GROUP_ID" --create-home \
             --home-dir /home/builder builder --groups sudo,dialout
 
+    # Shellcheck uses Docker. Docker socket may not be accessible due to
+    # permissions, so make docker binary setuid.
+    chmod +s "$(which docker)"
```

Comment:
It is possible to skip adding setuid to keep permissions more constrained:
- read GID of docker.sock (e.g. `stat -c "%g" /var/run/docker.sock`)
- create group for that GID (only if not present), normally this should be `docker` group but GID for that group may be different in container than it is on host
- add `builder` user to that group

Reply:
Shellcheck is now installed in this container and we no longer access host's Docker.